auth/credentials/internal/externalaccount/aws_provider.go

```diff
@@ -94,32 +94,30 @@ func (sp *awsSubjectProvider) subjectToken(ctx context.Context) (string, error)
 	if sp.RegionalCredVerificationURL == "" {
 		sp.RegionalCredVerificationURL = defaultRegionalCredentialVerificationURL
 	}
-	if sp.requestSigner == nil {
-		headers := make(map[string]string)
-		if sp.shouldUseMetadataServer() {
-			awsSessionToken, err := sp.getAWSSessionToken(ctx)
-			if err != nil {
-				return "", err
-			}
-
-			if awsSessionToken != "" {
-				headers[awsIMDSv2SessionTokenHeader] = awsSessionToken
-			}
-		}
-
-		awsSecurityCredentials, err := sp.getSecurityCredentials(ctx, headers)
+	headers := make(map[string]string)
+	if sp.shouldUseMetadataServer() {
+		awsSessionToken, err := sp.getAWSSessionToken(ctx)
 		if err != nil {
 			return "", err
 		}
-		if sp.region, err = sp.getRegion(ctx, headers); err != nil {
-			return "", err
-		}
-		sp.requestSigner = &awsRequestSigner{
-			RegionName:             sp.region,
-			AwsSecurityCredentials: awsSecurityCredentials,
+
+		if awsSessionToken != "" {
+			headers[awsIMDSv2SessionTokenHeader] = awsSessionToken
 		}
 	}
 
+	awsSecurityCredentials, err := sp.getSecurityCredentials(ctx, headers)
+	if err != nil {
+		return "", err
+	}
+	if sp.region, err = sp.getRegion(ctx, headers); err != nil {
+		return "", err
+	}
+	sp.requestSigner = &awsRequestSigner{
+		RegionName:             sp.region,
```

Comment:
@aeitzman In the related issue, you wrote:

>  Instead of completely rebuilding the request signer, it might make more sense to retrieve a new security credential and modify requestSigner.AwsSecurityCredentials, that would better match the implementation in other languages.I can make a PR for that fix.

What happened to this plan? Is it not desirable or practical to cache the result of `getRegion`?

Reply:
I was mistaken when I made the earlier comment, we actually do grab the region again in other languages isntead of caching it (see https://github.com/googleapis/google-auth-library-java/blob/afe4ff9f1f3c406c72ecb66fba50d7463f620467/oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java#L134C1-L135C1)

Reply:
Additionally, now that we have the custom provider as an option for AWS, it would be confusing if we re-call the provider for the security credentials each time, but not the region. It's possible some implementations of the provider could handle multiple credentials/regions based on the options/context getting passed into it